### PR TITLE
use markdown marker

### LINKED ISSUE #26: [Proposal] remove ---- xxx ---- markers, user markdown headlines in chat mode
Replace usages of:

-  `"------ User ------\n"` marker with `# > User:`
-  `"------ Assistant ------\n"` with `## > Assistant:`

Example:

``` md
# > User: hello

## > Assistant: Hello! How can I assist you today?

```

Benefit:
- nice presentation, similar to markdown 
- built in support for imenu, 
- can hide and show chat nodes

I have a prototype here https://github.com/thanhvg/mind-wave/commit/4995c7f23251139ad891c5640af78a7bd63945d2 if you think it's a good idea I can create a PR.

Thanks.

### PR DESCRIPTION
**Pull request by thanhvg**
close #26

- breaking change, will not work with old chat files.
- there is a bug with `imenu` for markdown mode with native-comp. Refer to https://github.com/jrblevin/markdown-mode/issues/578 for a workaround.

### CODE REVIEW

File: mind-wave.el [hunk 1]
```diff
@@ -341,7 +341,7 @@ Then Mind-Wave will start by gdb, please send new issue with `*mind-wave*' buffe
     (define-key map (kbd "C-,")   #'mind-wave-chat-ask-insert-line)
     (define-key map (kbd "C-S-m") #'mind-wave-chat-ask-send-buffer)
     (define-key map (kbd "C-u")   #'mind-wave-chat-continue)
-    (define-key map (kbd "C-i")   #'mind-wave-chat-generate-title)
+    (define-key map (kbd "C-c C-i")   #'mind-wave-chat-generate-title)
```
Comment (thanhvg) [inline mind-wave.el:344]: Current binding takes over my tab key. I can remove this line if necessary though.

**Comment by manateelazycat** [comment]
We should add convert function before ```mind-wave-call-async "chat_ask"``` to help user convert old marker to new marker.

File: mind-wave.el (hunk 1 above)
Comment (manateelazycat) [reply mind-wave.el:344]: That's ok since mind-wave generate title automatically, this binding is less use.

File: mind_wave.py [hunk 2]
```diff
@@ -159,11 +158,14 @@ def chat_parse_content(self, buffer_content):
         content = ''  # initialize the content
 
         for line in lines:
-            if line.startswith('------ '):
+            if line.startswith('# > ') or line.startswith('## > '):
```
Comment (manateelazycat) [inline mind_wave.py:161]: Should use 

```
if line == "# > User:" or line == "## > Assistant:":
```

**Review by manateelazycat — CHANGES_REQUESTED** [review]

File: mind_wave.py (hunk 2 above)
Comment (thanhvg) [reply mind_wave.py:161]: I prefer chat content is right after the markers, not on a new line. With this, we can still see the first line of chat when we collapse the nodes:

Example:
- full display
image
- chat nodes collapsed
image
- all collapses
image
- imenu
image

Comment (manateelazycat) [reply mind_wave.py:161]: Please add convert function before mind-wave-call-async "chat_ask" to help user convert old marker to new marker.

This patch make old *.chat file can't work, i won't merge this patch if convert function not provide, sorry.

Comment (thanhvg) [reply mind_wave.py:161]: sure I will do it when I have time, thanks

Comment (thanhvg) [reply mind_wave.py:161]: done, added a convert function and and a flag to run it.
then the flag is true which is the default value, the function gets run on chat mode enabled and insde `mind-wave-chat-ask-send-buffer`

**Pull request merged**